mlir/lib/Conversion/ArithToEmitC/ArithToEmitC.cpp

```diff
@@ -757,7 +757,9 @@ class TruncFConversion : public OpConversionPattern<arith::TruncFOp> {
       return rewriter.notifyMatchFailure(castOp,
                                          "unsupported cast destination type");
 
-    if (!castOp.areCastCompatible(operandType, dstType))
+    if (!isa<emitc::OpaqueType>(dstType) &&
+        !isa<emitc::OpaqueType>(operandType) &&
+        !castOp.areCastCompatible(operandType, dstType))
```

Comment:
Are there other places where `areCastCompatible` is called? Like in the verifier? 
I though we would need to change `areCastCompatible` itself to say `true` if any involved type was opaque.

Reply:
The arith op is asked for cast compatibility, but I guess that is actually wrong. We should query the EmitC cast-compatibility. Changed it here: https://github.com/Xilinx/llvm-project/pull/428/commits/96d2f928e5b1f3f553f1c354ef7aa12922c95d28